src/upload_node.py

```diff
@@ -0,0 +1,117 @@
+from dataclasses import dataclass
+
+import rospy
+
+from sensor_msgs.msg import NavSatFix
+from std_msgs.msg import Float64
+
+from martian_mines.msg import FigureMsgList, FigureMsg
+from utils.coords_scaler import CoordinateScaler
+from utils.uploader import Uploader
+
+
+@dataclass
+class DroneData:
+    global_pose: NavSatFix = None
+    heading: Float64 = None
+    rel_altitude: Float64 = None
+
+    def is_complete(self) -> bool:
+        return all(param is not None for param in (self.global_pose, self.heading, self.rel_altitude))
+
+    def reset(self) -> None:
+        self.global_pose = None
+        self.heading = None
+        self.rel_altitude = None
```

Comment:
Use [message_filters](http://wiki.ros.org/message_filters) to get one callback from multiple subscribers, you can see example at figure finder node

Reply:
I used ApproximateTimeSynchronizser, because we get Float64 messages which are headerless